apps/etterlatte-trygdetid/src/main/kotlin/trygdetid/Trygdetid.kt

```diff
@@ -98,4 +103,10 @@ data class TrygdetidPeriode(
     init {
         require(fra.isBefore(til) || fra.isEqual(til)) { "Ugyldig periode, fra må være før eller lik til" }
     }
-}
\ No newline at end of file
+
+    fun overlapperMed(other: TrygdetidPeriode): Boolean {
+        return this.fra.isBefore(other.til) && other.fra.isBefore(this.til)
```

Comment:
Kjens som denne også burde sjekke motsatt veg, typ

`        return (this.fra.isBefore(other.til) && other.fra.isBefore(this.til))
|| (other.fra.isBefore(this.til) && this.fra.isBefore(other.til))`?

Reply:
You could play with it :)

`this.fra.datesUntil(this.til.plusDays(1)).intersect(other.fra.datesUntil(other.til.plusDays(1))).isNotEmpty()`

Reply:
Or in slightly more readable form:

```
fun LocalDate.datesUntilInclusive(date: LocalDate) = this.datesUntil(date.plusDays(1)).toList()
fun List<LocalDate>.overlap(other: List<LocalDate>) = this.intersect(other.toSet()).isNotEmpty()

val rangeA = this.fra.datesUntilInclusive(this.til)
val rangeB = other.fra.datesUntilInclusive(other.til)

rangeA.overlap(rangeB)
```

But this is probably not the "optimal" solution :)

Reply:
Trenger man det? 
La til flere tester nå for å sjekke ulike varianter. Ser ut som det stemmer?
[b71d426](https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/1593/commits/b71d426bb7ffae6417651c88f2d7921a41d1ea1e)

Reply:
Du trenger ikke mitt forslag - det var mest en liten divergering :)